crates/storage/provider/src/providers/snapshot/writer.rs

```diff
@@ -228,8 +218,6 @@ impl<'a> SnapshotProviderRW<'a> {
     ) -> ProviderResult<BlockNumber> {
         debug_assert!(self.writer.user_header().segment() == SnapshotSegment::Headers);
 
-        self.writer.user_header_mut().increment_block();
```

Comment:
Are we not supposed to increment the block here as we append a new header?

Reply:
increment is done now outside of this method, just as for `append_receipt` and `append_transaction` methods: https://github.com/paradigmxyz/reth/blob/ee9265b7007c5dd891745977262167d3bffb5fc4/crates/snapshot/src/segments/headers.rs#L52-L55

@joshieDo has some improvements to `append_header` method in https://github.com/paradigmxyz/reth/pull/6273, will re-use these changes once his PR is merged.